### ISSUE TITLE
False SC2154 positive when assigning a variable via mapfile/readarray

### ISSUE THREAD

**Issue by 64kramsystem** [koalaman/shellcheck#2719]
#### For bugs
- Rule Id: SC2154
- My shellcheck version: 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [?] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

mapfile -td'' myvar < <(find /tmp -name test -print0)
echo "${myvar[@]}"
```

#### Here's what shellcheck currently says:
```
Line 4:
echo "${myvar[@]}"
      ^-- [SC2154](https://www.shellcheck.net/wiki/SC2154) (warning): myvar is referenced but not assigned.

```

#### Here's what I wanted or expected to see:
(no warnings)
